gobblin-utility/src/test/java/gobblin/util/AvroUtilsTest.java

```diff
@@ -242,6 +244,56 @@ public void testSwitchName() {
     GenericRecord record = (GenericRecord) reader.read(oldRecord, decoder);
 
     Assert.assertEquals(AvroUtils.getFieldValue(record, TEST_FIELD1_LOCATION).get().toString(), TEST_VALUE1);
+    Assert.assertEquals(AvroUtils.getFieldSchema(record.getSchema(), TEST_FIELD1_LOCATION).get().getType(),
+        Schema.Type.STRING);
     Assert.assertEquals(AvroUtils.getFieldValue(record, TEST_FIELD2_LOCATION).get().toString(), TEST_VALUE2);
+    Assert.assertEquals(AvroUtils.getFieldSchema(record.getSchema(), TEST_FIELD2_LOCATION).get().getType(),
+        Schema.Type.STRING);
+  }
+
+  @Test(expectedExceptions = AvroRuntimeException.class)
+  public void testComplexTypesInUnionNotSupported() {
+    final String TEST_OBJECT = "testUnionObject";
+    final String TEST_FIELD_LOCATION = TEST_OBJECT + "." + this.TEST_FIELD1;
+
+
+    String testSchema = "{" +
```

Comment:
Since the schema is getting bigger, can you please move it to a resource file and read from it (refer to gobblin-data-management test cases for quick reference of tests that do that)